cirq-google/cirq_google/engine/qcs_notebook.py

```diff
@@ -134,7 +134,12 @@ def get_qcs_objects_for_notebook(
         )
         if not processors:
             raise ValueError("No processors available.")
-        processor = processors[0]
+        if virtual:
+            processor = processors[0]
+        else:
+            if not any(p.health() == 'OK' for p in processors):
+                raise ValueError("No healthy processors available.")
+            processor = next(p for p in processors if p.health() == 'OK')
```

Comment:
I think we can remove the branching here and require there is a healthy processor whether virtual or not.
The exceptions at lines 136, 141 could be combined, ie, can we remove the lines 135-139, and only keep 140-142?